### PR TITLE
Fix icc / cub

### PR DESCRIPTION
**Pull request by gevtushenko**
## Description

closes https://github.com/NVIDIA/cccl/issues/1151

Suppresses visibility warning for icc. 

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by wmaxey — APPROVED** [review]

File: cub/cub/util_macro.cuh [hunk 1]
```diff
@@ -137,6 +137,9 @@ _CCCL_DIAG_SUPPRESS_CLANG("-Wattributes")
 #if !defined(_CCCL_CUDA_COMPILER_NVHPC)
 _CCCL_DIAG_SUPPRESS_NVHPC(attribute_requires_external_linkage)
 #endif // !_CCCL_CUDA_COMPILER_NVHPC
+#if defined(__ICC)
```
Comment (miscco) [inline cub/cub/util_macro.cuh:140]: Important: this is technically not the right define, we should use our compiler detection instead
```suggestion
#if defined(_CCCL_COMPILER_ICC) || defined(_CCCL_COMPILER_ICC_LLVM)
```

File: cub/cub/util_macro.cuh [hunk 2]
```diff
@@ -137,6 +137,9 @@ _CCCL_DIAG_SUPPRESS_CLANG("-Wattributes")
...
+#pragma warning(disable: 1890)
```
Comment (miscco) [inline cub/cub/util_macro.cuh:141]: Important; Please add a comment what exact warning is suppressed here, its difficult to remember all those
```suggestion
#pragma warning(disable: 1890) // the "__visibility__" attribute can only appear on functions and variables with external linkage
```

**Review by miscco — CHANGES_REQUESTED** [review]

File: cub/cub/util_macro.cuh (hunk 1 above)
Comment (gevtushenko) [reply cub/cub/util_macro.cuh:140]:
```
cat main.cu 
#include <cuda/__cccl_config>
#include <iostream>

int main() {
#if defined(_CCCL_COMPILER_ICC) || defined(_CCCL_COMPILER_ICC_LLVM)
	std::cout << "defined(_CCCL_COMPILER_ICC) || defined(_CCCL_COMPILER_ICC_LLVM)" << std::endl;
#endif

#if defined(__ICC)
	std::cout << "defined(__ICC)" << std::endl;
#endif
}


/usr/local/cuda/bin/nvcc -ccbin=icc main.cu -I/cccl/libcudacxx/include


./a.out 
defined(__ICC)
```

File: cub/cub/util_macro.cuh (hunk 2 above)
Comment (gevtushenko) [reply cub/cub/util_macro.cuh:141]: The suppression is under `#if !defined(CUB_DISABLE_KERNEL_VISIBILITY_WARNING_SUPPRESSION)`, which seems to indicate the purpose.

Comment (miscco) [reply cub/cub/util_macro.cuh:141]: Does it hurt to add it? I really want to know what `#pragma warning` does without looking at surrounding code

**Review by miscco — APPROVED** [review]

**Pull request merged**